### PR TITLE
pscanrules alpha/beta: Promote 'Modern Web Application' rule to Beta

### PR DESCRIPTION
**Pull request by kingthorin**
- Moved rule class.
- Moved unit test.
- Moved resource keys/values.
- Moved help entry.

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

File: addOns/pscanrulesBeta/src/main/javahelp/org/zaproxy/zap/extension/pscanrulesBeta/resources/help/contents/pscanbeta.html [hunk 1]
```diff
@@ -65,6 +65,12 @@ <H2>HTTPS to HTTP Insecure Transition in Form Post</H2>
 <p>
 Latest code: <a href="https://github.com/zaproxy/zap-extensions/blob/master/addOns/pscanrulesBeta/src/main/java/org/zaproxy/zap/extension/pscanrulesBeta/InsecureFormPostScanner.java">InsecureFormPostScanner.java</a>
 
+<H2>Modern Web Application</H2>
+This raises an informational alert if a site appears to be a modern web application.<br>
+It does not indicate any potential vulnerabilities but it could indicate that the ajax spider might be more effective at exploring this site compared to the traditional spider.
+<p>
+Latest code: <a href="https://github.com/zaproxy/zap-extensions/blob/master/addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesBeta/ModernAppDetectionScanner.java">ModernAppDetectionScanner.java</a>
```
Comment (thc202) [inline addOns/pscanrulesBeta/src/main/javahelp/org/zaproxy/zap/extension/pscanrulesBeta/resources/help/contents/pscanbeta.html:72]: The ID of the add-on needs to be updated.

Comment (kingthorin) [reply addOns/pscanrulesBeta/src/main/javahelp/org/zaproxy/zap/extension/pscanrulesBeta/resources/help/contents/pscanbeta.html:72]: Oops got one and missed one

**Comment by kingthorin** [comment]
Fixed

**Comment by thc202** [comment]
Thanks!

**Pull request merged**